src/prefect/utilities/feature_flags.py

```diff
@@ -0,0 +1,171 @@
+                             if feature flagging is disabled.
+    """
+    if not settings.PREFECT_FEATURE_FLAGGING_ENABLED.value():
+        return
+
+    if not client:
+        client = get_feature_flag_client()
+
+    # If the flag exists in the feature flag store, we'll consider the
+    # enabled state, bucketer, and conditions currently saved in the
+    # feature flag store as canonical.
+    if client.exists(flag_name):
+        return client.get(flag_name)
+
+    flag = client.create(flag_name, is_enabled=is_enabled, client_data=client_data)
+
+    if bucketer:
+        flag.set_bucketer(bucketer)
+
+    if conditions:
+        flag.set_conditions(conditions)
+
+    return flag
+
+
+def flag_is_enabled(
+    flag_name: str,
+    default=False,
+    client: FeatureFlagClient = None,
+    **conditions: Optional[Any]
+):
+    """
+    Check if a feature flag is enabled.
+
+    This function always returns False if the setting
+    `PREFECT_CLOUD_ENABLE_FEATURE_FLAGGING` is false.
+
+    NOTE: If `flag_is_enabled()` is called for a feature that has conditions,
+    but the caller does not give any conditions, the current state of the flag
+    is returned.
+
+    Args:
+        flag_name: the name of the feature flag
+        default: the default return value to use if no feature flag with
+                 the given name exists. Defaults to False.
+        client: The `FeatureFlagClient` instance to use. Defaults to a client
+                configured to look at an in-memory feature store.
+        conditions: keyword arguments, e.g. `is_admin=True`, to check
+                    against any Conditions on the flag
+
+    Returns:
+        bool: whether the flag is enabled
+    """
+    if not settings.PREFECT_FEATURE_FLAGGING_ENABLED.value():
+        return False
+
+    if not client:
+        client = get_feature_flag_client()
+
+    return client.is_enabled(flag_name, default=default, **conditions)
+
+
+def list_feature_flags(
+    batch_size: int = 10, client: FeatureFlagClient = None
+) -> List[FeatureFlag]:
+    """
+    List all feature flags.
+
+    This function always returns an empty list if the setting
+    `PREFECT_CLOUD_ENABLE_FEATURE_FLAGGING` is false.
+
+    Args:
+        batch_size: batch size of flags to retrieve at a time
+        client: The `FeatureFlagClient` instance to use. Defaults to a client
+                configured to look at an in-memory feature store.
+
+    Returns:
+        List[FeatureFlag]: list of all feature flags in the store
+    """
+    if not settings.PREFECT_FEATURE_FLAGGING_ENABLED.value():
+        return []
+
+    if not client:
+        client = get_feature_flag_client()
+
+    flags = []
+    offset = 0
+
+    while True:
+        batch = list(client.list(limit=batch_size, offset=offset))
+
+        if not batch:
+            break
+
+        flags.extend(batch)
+        offset += batch_size
+
+    return flags
+
+
+# Feature flags currently in use go here.
```

Comment:
It would be nice if the definitions of flags could be in a separate module from the support functions.  I don't have a great idea for how to organize them, though...

We want the flags to be registered automatically at any point we would want to check them, and having them at the bottom of this module makes that true (you have to import this module to get the functions to check them).  Is there another way we could organize things and preserve that?

Reply:
Yeah, I'm thinking about that. I already need to reorganize this a bit because I want to use a separate client instance for testing, which with my "simple functions" approach now means threading that through every call.

Using module side-effects is an antipattern, but it works in Cloud where we can be super minimal. But ultimately, the code in OSS should support changing out the "store" so that our default and simple OSS implementation is easy to replace with the Redis store we use in Cloud -- while all the feature flags and code paths in OSS continue to work and can be toggled/tested in Cloud.